storage/bucket.go

```diff
@@ -515,6 +515,8 @@ type BucketAttrsToUpdate struct {
 	// is provided, the value is ignored in the update.
 	CORS []CORS
 
+	Lifecycle Lifecycle
```

Comment:
The problem with this is that we don't whether the user wants to update to this Lifecycle, or not update the Lifecycle at all. So the type should be *Lifecycle.

Reply:
Unfortunately the toRawLifecycle function does not accept pointers and i am just re-using the same code from BucketAttrs. I can change this to a pointer but will need to change this function which will have a knock on effect with BucketAttrs. Let me know how you want me to proceed. I can get the changes done today.